security-privacy.html

```diff
@@ -207,90 +207,117 @@
+          writing.
+          The preferred type of this NDEF record is NFC Forum External Type
+          (TNF=0x04), with the External Type field set to
+          <code>urn:nfc:ext:w3.org:web-nfc-bl-write</code>.
+          The payload is a string list of URL's, delimited by white space.
+        </li>
+      </ul>
+    </p>
+  </section>
+  <section> <h3>Web NFC using encoded metadata</h3>
     <p>
-      For making tags that can be read and written by the Web NFC API, and for
-      making sure that web pages cannot access generic NFC tags, the following
-      are possible:
+      Web NFC may also define an own format for accepted NDEF records,
+      where the preferred type of this NDEF record is NFC Forum External Type
+      (TNF=0x04), with the External Type field set to
+      <code>urn:nfc:ext:w3.org:webnfc</code>. The payload type would be
+      <code>application/webnfc+json</code> (by default UTF-8 encoded). All data
+      would be JSON, for instance with the following schema:
       <ul>
-        <li>Use the NDEF record <code>id</code> field in order to store the
-          origin of the page creating the record, which is also by default
-          allowed to modify the tag.</li>
-        <li>Use additional special NDEF records in each Web NFC tag, which are
-          hidden by the implementation. This solution depends on support for
-          writing multiple NDEF records per NDEF message.</li>
-        <li>Use additional data encoded into the NFC payload in a way which is
-          specific to Web NFC API. This solution is not desirable.</li>
+        <li>
+          <code>id</code>: <code>"string"</code>; contains the Web NFC
+          identifier, e.g. the SHA256 signature of <code>"w3.org/web-nfc”</code>
+        </li>
+        <li>
+          <code>scope</code>: <code>"string"</code>; contains the origin of the
+          data; it can be also stored at the NDEF record's
+          <code>id</code> field.
+        </li>
+        <li>
+          <code>messageType</code>: <code>"string"</code>; contains the type of
+          the data, either <code>"text"</code>, <code>"url"</code>,
+          <code>"json"</code>, or <code>"blob"</code>.
+        </li>
+        <li>
+          <code>message</code>: <code>"string"</code>; contains the payload data
+          encoded as text (text, URL and JSON), and for blob encoded as Base64.
+        </li>
+        <li>
+          <code>whitelistRead</code>: <code>"array"</code>; contains the URL
+          scopes that are explicitly allowed to read the data. No other scope
+          is permitted to read the data.
+        </li>
+        <li>
+          <code>whitelistWrite</code>: <code>"array"</code>; contains the URL
+          scopes that are explicitly allowed to rewrite the data. No other scope
+          is permitted to rewrite (update) the data.
+        </li>
+        <li>
+          <code>blacklistRead</code>: <code>"array"</code>; contains the URL
```

Comment:
I find a blacklist useless... it is so easy to work around... put an iframe to a different url, read and send info back. Let's not fool people into believe it is safe